tests/core/pyspec/eth2spec/test/helpers/execution_payload.py

```diff
@@ -0,0 +1,26 @@
+
+def build_empty_execution_payload(spec, state):
+    """
+    Assuming a pre-state of the same slot, build a valid ExecutionPayload without any transactions.
```

Comment:
I think there's a subtle bug in common usage here.

1. `build_empty_block_for_next_slot` returns `build_empty_block(spec, state, state.slot + 1)`
2. `build_empty_block` if a param is passed in for `slot`, does *not* pass this along to  `build_empty_execution_payload` which just assumes the slot from `state`
3. `timestamp` is computed from `state.slot`
4. Thus, `block.slot` and `block.execution_payload.timestamp` will have a mismatch when using `build_empty_block_for_next_slot` (or if you pass in any non-current-slot value to `build_empty_block`

Immediate solutions: either transition a dummy state forward to the slot in question within `build_empty_block` or pass in `slot` to `build_empty_execution_payload`

Reply:
Are we not hitting this in `process_execution_payload`:
```python
assert execution_payload.timestamp == compute_time_at_slot(state, state.slot)
```

Seems that either (1) my bug report is incorrect, (2) we never really use `build_empty_block_for_next_slot`, or (3) we are never actually running `process_execution_payload`.

We can add a sanity condition to the `process_execution_payload` tests to see if the execution header changed to narrow it down.

Reply:
` build_empty_execution_payload` in `build_empty_block_for_next_slot` uses a copy of `state` which is already transitioned to that same slot.

Reply:
indeed it is. whoops!